File: main/kotlin/game/model/elements/locations/Location.java
package game.model.elements.locations;

import game.model.basics.Game;
import game.model.constants.HomeStatus;
import game.model.constants.StoryStatus;
import game.model.elements.sections.Section;

import java.util.ArrayList;

import static game.model.constants.Messages.INCORRECT;

public abstract class Location {
    public String name;
    public ArrayList<Section> sections;
    public final ArrayList<String> sectionsId = new ArrayList<>();

    public Location(String name) {
        this.name = name;
        sections = new ArrayList<>();
    }

    public String interaction(String[] command) {
        for (int i = 0; i < sections.size(); i++) {
            if (Game.status.equals(sections.get(i).SECTION_NAME)) {
                return sections.get(i).interaction(command);
            }
        }

        if (Game.status.equals(StoryStatus.INTRO)) {
            for (int i = 0; i < sections.size(); i++) {
                if (sections.get(i).SECTION_NAME.equals(HomeStatus.BEDROOM)) {
                    return sections.get(i).interaction(command);
                }
            }
        }

        return INCORRECT;
    }

}

File: src/main/kotlin/model/elements/Location.kt
package model.elements

import model.db.DB
import model.elements.Section.Companion.fillSections

interface Location {
    val sections: ArrayList<Section>
    val key: String
    val name: String


    companion object {
        const val HOME = "Home"
        const val STREET = "Street"

        fun location(name: String): Location {
            val dbLocation = DB.loadLocationFromDB(name)

            return object : Location {
                override val key = dbLocation.key!!
                override val name = dbLocation.name!!
                override val sections = fillSections(dbLocation)
            }
        }
    }
}